File: MyFinal/app/src/main/java/com/example/mfinal/MainActivity.java
package com.example.mfinal;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.example.mfinal.mvp.presenter.MainPresenter;
import com.example.mfinal.mvp.view.MainView;
import com.example.mfinal.ui.BackButtonListener;

import javax.inject.Inject;

import moxy.MvpAppCompatActivity;
import moxy.presenter.InjectPresenter;
import ru.terrakok.cicerone.Navigator;
import ru.terrakok.cicerone.NavigatorHolder;
import ru.terrakok.cicerone.android.support.SupportAppNavigator;

public class MainActivity extends MvpAppCompatActivity implements MainView {

    @InjectPresenter
    MainPresenter presenter;

    @Inject
    NavigatorHolder navigatorHolder;

    private Navigator navigator = new SupportAppNavigator(this, getSupportFragmentManager(), R.id.container);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        CovidStatsApp.INSTANCE.getAppComponent().inject(this);

    }

    @Override
    protected void onResumeFragments() {
        super.onResumeFragments();
        navigatorHolder.setNavigator(navigator);
    }

    @Override
    protected void onPause() {
        super.onPause();
        navigatorHolder.removeNavigator();
    }


    @Override
    public void onBackPressed() {
//        super.onBackPressed();

        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (fragment instanceof BackButtonListener && ((BackButtonListener)fragment).backPressed()) {
                return;
            }
        }

        presenter.backClicked();
    }
}

File: Kotlin/app/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val button = findViewById<Button>(R.id.button)
        val nativeColor = button.currentTextColor
        button.setOnClickListener { if (button.currentTextColor != Color.YELLOW) button.setTextColor(Color.YELLOW) else button.setTextColor(nativeColor) }
    }
}